etl/jobs/run_change_detection_batches.py

```diff
@@ -0,0 +1,158 @@
+      if not response or not response["Workflow"]: exit("Could not retrieve Workflow with name " + workflow_name)
+      status = response["Workflow"]["LastRun"]["Status"]
+      if status in ['RUNNING', 'STOPPING']:
+         time.sleep(60)
+         continue
+      elif status in ['COMPLETED', 'ERROR', 'STOPPED']:
+         break
+      else:
+         exit("Encountered unexpected workflow status: " + status)  
+
+def await_workflow_run_completion(client, workflow_name, run_id):
+   while True:
+      response = client.get_workflow_run(
+          Name=workflow_name,
+          RunId=run_id,
+      )
+      status = response["Run"]["Status"]
+      if status in ['RUNNING', 'STOPPING']:
+         time.sleep(60)
+         continue
+      elif status == 'COMPLETED':
+         break
+      else:
+         exit("Encountered unexpected workflow status: " + status)  
+
+def get_trigger(client, trigger_name):
+   triggerResponse = client.get_trigger(
+      Name=trigger_name
+   )
+   if not triggerResponse or not triggerResponse["Trigger"]: exit("Could not retrieve Trigger with name " + trigger_name)
+   return triggerResponse["Trigger"]
+
+def set_triggered_job_params(client, trigger, job_name, params):
+   for x in trigger["Actions"]:
+      if x['JobName'] == job_name:
+         arguments = x['Arguments']
+         break
+   if not arguments: exit("Could not locate arguments for job " + job_name)
+   for key,value in params.items():      
+      arguments[key] = value      
+
+   # update_trigger API properties are a subset of get_trigger properties
+   # and we have to provide exactly all of those (can neither include non-updateable properties nor exclude mandatory updateable properties which have not been updated)
+   trigger_update_params =  {
+         'Name': trigger["Name"],
+         'Description': trigger.get("Description", ''),
+         'Schedule': trigger.get("Schedule", ''),
+         'Actions': trigger["Actions"],
+         'Predicate': trigger["Predicate"],
+   }
+   if trigger.get("EventBatchingCondition") is not None:
+      trigger_update_params['EventBatchingCondition'] = trigger["EventBatchingCondition"]
+   
+   client.update_trigger(
+       Name=trigger["Name"],
+       TriggerUpdate=trigger_update_params
+   )
+
+def start_workflow(client, workflow_name):
+   response = client.start_workflow_run(
+       Name=workflow_name
+   )
+   return response["RunId"] 
+
+args = getResolvedOptions(sys.argv, ['JOB_NAME', 'CONTEXT', 'FROM_YEAR', 'FROM_MONTH', 'TO_YEAR', 'TO_MONTH', 'MONTHS_PER_RUN'])
+
+sc = SparkContext()
+glueContext = GlueContext(sc)
+spark = glueContext.spark_session
+job = Job(glueContext)
+job.init(args['JOB_NAME'], args)
+
+context = contexts.get(args['CONTEXT'])
+if not context: exit("No context parameters found for supplied parameter " + args['CONTEXT'])
+date_ranges = get_date_ranges(int(args['FROM_YEAR']), int(args['FROM_MONTH']), int(args['TO_YEAR']), int(args['TO_MONTH']), int(args['MONTHS_PER_RUN']))
```

Comment:
If we're expecting users to run this then it might be worth adding in explicit validation. E.g. FROM_YEAR <= TO_YEAR and that months are always 1-12

Reply:
Good idea, thanks, will do that.  Currently if they got anything wrong like FROM_YEAR > TO_YEAR the get_date_ranges function would spit out a generic "From and To date ranges not specified correctly" and they should be able to work out what they've done wrong, but more explicit validation would be more user friendly and I hadn't thought about month not between 1 and 12

Reply:
Now done